src/main/java/com/commercetools/project/sync/service/impl/ReferencesServiceImpl.java

```diff
@@ -69,53 +66,53 @@ public ReferencesServiceImpl(@Nonnull final SphereClient ctpClient) {
       return fetchCustomObjectKeys(customObjectIds);
     }
 
-    // TODO: Make sure each nonCached set has less than 500 resources, otherwise batch requests
-    // https://github.com/commercetools/commercetools-project-sync/issues/42
-    final CombinedResourceKeysRequest combinedResourceKeysRequest =
-        new CombinedResourceKeysRequest(
-            nonCachedProductIds, nonCachedCategoryIds, nonCachedProductTypeIds);
+    /*
+     * An id is a 36 characters long string. (i.e: 53c4a8b4-754f-4b95-b6f2-3e1e70e3d0c3) We
+     * chunk them in 300 ids we will have around a query around 11.000 characters. Above this size it
+     * could return - Error 413 (Request Entity Too Large)
+     */
+    final int CHUNK_SIZE = 300;
```

Comment:
As it's used in 2 different methods this might be an instance variable on the class level ?

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/234/commits/e2db35dbb6e86c8f9ca45cd386402655200093ca